### PR TITLE
Extract lowlevel bits into a `core` package

### PR DESCRIPTION
**Pull request by clavoie-takt**
Should be self-explanatory -- though the nix build bits are unchecked, they didn't work on my machine before or after. Would love some guidance on whether they're supposed to work or are a WIP, and how to approach nix in general.

### CODE REVIEW

File: core/tests/UnsafeTests.hs [hunk 1]
```diff
@@ -0,0 +1,203 @@
+{-# LANGUAGE OverloadedStrings   #-}
```
Comment (Gabriel439) [inline core/tests/UnsafeTests.hs]: Is the `./tests/UnsafeTests.hs` file in the parent project still needed?

**Comment by clavoie-takt** [comment]
Yeah, a while tmp directory got included accidentally -- nuked. Fixed the duplicate test too.

**Comment by Gabriel439** [comment]
Yeah, this looks great!  I will see if we can fix the Nix issue before merging, but if we can't do so within a few days then we can go ahead and merge this anyway

**Comment by clavoie-takt** [comment]
Gentle ping @Gabriel439 -- any updates on making it work with nix?

**Comment by intractable** [comment]
I'll try to take a look at this tomorrow, sorry for the delay.

**Comment by Gabriel439** [comment]
@clavoie-takt: I updated the Nix build logic in this branch based off of yours:

https://github.com/awakesecurity/gRPC-haskell/tree/gabriel/wip

I had to first upgrade Nixpkgs to 17.09 so I need to first wait for https://github.com/awakesecurity/gRPC-haskell/pull/45 to be reviewed but in the meantime you can merge my branch into yours

**Comment by clavoie-takt** [comment]
Done. Tests pass on my box (incl. through nix), I think we're good to go.

**Comment by Gabriel439** [comment]
@clavoie-takt: It looks like there is still a merge conflict to address

**Comment by clavoie-takt** [comment]
Forgot a push, a personal classic. Fixed :)

**Comment by Gabriel439** [comment]
Awesome!  Thank you for doing this :)

**Pull request merged**